components/playlist/playlist_service.cc

```diff
@@ -563,7 +616,53 @@ void PlaylistService::OnGetOrphanedPaths(
   }
 }
 
-void PlaylistService::CleanUp() {
+void PlaylistService::CleanUpMalformedPlaylistItems() {
```

Comment:
nit: I think we can nuke all existing playlist data if it includes old format instead of using filtering logic below.
I don't think playlist data could have old & new format together.

Reply:
Yes, that'd be simpler! Done.